cpp/src/arrow/compute/kernels/vector_cumulative_ops.cc

```diff
@@ -217,53 +253,142 @@ const FunctionDoc cumulative_min_doc{
      "start as the new minimum)."),
     {"values"},
     "CumulativeOptions"};
+
+const FunctionDoc cumulative_mean_doc{
+    "Compute the cumulative mean over a numeric input",
+    ("`values` must be numeric. Return an array/chunked array which is the\n"
+     "cumulative mean computed over `values`. CumulativeOptions::start_value is \n"
+     "ignored."),
+    {"values"},
+    "CumulativeOptions"};
 }  // namespace
```

Comment:
The anonymous namespace could encompass more symbols, for example below.

Reply:
Done